File: hackerrank/simple_array_sum/simple_array_sum.go
package main
import (
    "fmt"
)

func main() {
 //Enter your code here. Read input from STDIN. Print output to STDOUT
    var times, input_array, result int
    
    _, err := fmt.Scan(&times)
    
    if err != nil {
        panic(err)
    }

    for i := 0; i < times; i++ {
        input_array = read()
        result += input_array
    }

    fmt.Println(result)

}

func read() int {
    var n int
    _, err := fmt.Scan(&n)

    if err != nil {
        panic(err)
    }

    return n
}

File: hackerrank/simple_array_sum/simple_array_sum.ex
defmodule Solution do
#Enter your code here. Read input from STDIN. Print output to STDOUT
   def sum() do
   		times = IO.gets("") |> String.trim |> String.to_integer
   		result = IO.gets("") |> 
   		String.trim |> 
   		String.split("") |> 
   		Enum.map(fn(x) -> String.to_integer(x) end) |>
   		Enum.take(times) |>
   		Enum.reduce(0, fn(x, acc) -> x + acc end)
   		IO.puts result
   end 
end
Solution.sum